internal/filter/filter.go

```diff
@@ -18,18 +18,60 @@
 package filter
 
 import (
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/sirupsen/logrus"
 	"open-match.dev/open-match/pkg/pb"
 )
 
 var emptySearchFields = &pb.SearchFields{}
 
-// InPool returns whether the ticket meets all the criteria of the pool.
-func InPool(ticket *pb.Ticket, pool *pb.Pool) bool {
+var (
+	logger = logrus.WithFields(logrus.Fields{
+		"app":       "openmatch",
+		"component": "filter",
+	})
+)
+
+// Criteria specifies all the criteria that the Ticket needs to meet to
+// be selected. This is used instead of the Pool directly to avoid any
+// criteria validation logic in the filter package.
+type Criteria struct {
+	DoubleRangeFilters  []*pb.DoubleRangeFilter
+	StringEqualsFilters []*pb.StringEqualsFilter
+	TagPresentFilters   []*pb.TagPresentFilter
+	CreatedBefore       time.Time
+	CreatedAfter        time.Time
+}
+
+// MeetsCriteria returns whether the ticket meets all the criteria of the pool.
+func MeetsCriteria(ticket *pb.Ticket, fc *Criteria) bool {
 	s := ticket.GetSearchFields()
 	if s == nil {
 		s = emptySearchFields
 	}
-	for _, f := range pool.GetDoubleRangeFilters() {
+
+	// CreateTime is only populated by Open Match and hence expected to be valid.
+	if ct, err := ptypes.Timestamp(ticket.CreateTime); err == nil {
```

Comment:
I don't quite like how this needs to be parsed every time.  It will be parsed for every ticket for every query, regardless of if it uses CreatedBefore / CreatedAfter.

We're setting this field, so it **should** be valid always.  Hm.
Maybe we need a type for a parsed ticket too... I don't know.  What are your thoughts?

If we are logging, it should be an error, because the system is definitely not behaving at any point where a field it set is invalid.

Reply:
Ok, possible course of action (unsure):
Create another type in this package that is created from a ticket proto.  No visible fields, but it's passed to the In/Meets method.  Maybe it can return the ticket that it was created from.

Then in query cache, use this object in the map instead of tickets.

If the ticket create time parsing fails, I think a reasonable thing to do is loudly log (but this will only happen when query cache reads the ticket, instead of every query), and then assume the ticket is one minute old.

Reply:
For your logging comment, I ve changed the log to Error. 

I see your concern around Ticket parsing. I did go down the path of exploring a type for the Ticket and updating query cache with it. Now, instead of parsing time on each query, you would end up re-creating the ticket for each query. Was not great. The alternative was for that custom type to just have (time.time and Ticket), store and use the entire ticket as is and only parse the time - so query cache still uses the custom type - but internally the type just stores Ticket+time. Seems strange again.

I've added a condition now so that the time gets parsed only if using before / after  - so slight improvement over the previous. The time parsing btw, is just another proto conversion - I dont think it would impact performance but again, I ve not validated that. With the condition though, atleast the impact is limited to only using the time range feature.

I would prefer to keep this change as is - and if we want to explore optimizing the time parsing, doing that as a separate change - given that is an internal only optimization.